### PR TITLE
Add sriov_config role

### PR DESCRIPTION
**Pull request by tonyskapunk**
This role takes care of configuring sriov node policies and/or networks. Requires sriov operator installed in the cluster and sriov capable NICs.

### CODE REVIEW

**Review by manurodriguez — DISMISSED** [review]
LGTM, just two minor suggestions in the README

File: roles/sriov_config/README.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+> - It's possible to omit a SR-IOV `network` when only a `node_policy` is applied.
+> - It's possible to omit a `node_policy` when the SR-IOV `network` to apply is mapped to an existing SR-IOV Network Node Policy
+
+#### SR-IOV Node Policy
+
+| Variable         | Required | Type    | Description
+| ---------------- | -------- | ------- | -----------
+| name             | Yes      | String  | Name of the SR-IOV Network Node Policy.
+| device_type      | No       | String  | The driver type for configured VFs. Allowed value "netdevice", "vfio-pci".
+| eswitch_mode     | No       | String  | NIC Device Mode. Allowed value "legacy","switchdev".
```
[the remark this answers is not in the document]
Comment (tonyskapunk) [reply roles/sriov_config/README.md:39]: :smile:  Looks like a typo but is actually the name [eSwtichMode](https://github.com/openshift/sriov-network-operator/blob/58986501f2b4f95e1612afe0c2b9ea9fb668fba8/manifests/stable/sriovnetwork.openshift.io_sriovnetworknodepolicies.yaml#L45).

But it made me realize that I should've named it  `e_switch_mode` instead.

File: roles/sriov_config/README.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+| num_vfs          | Yes      | Int     | Number of VFs for each PF
+| priority         | No       | Int     | Priority of the policy, higher priority policies can override lower ones
+| vdpa_type        | No       | String  | VDPA device type. Allowed value "virtio", "vhost"
+
+#### NIC Selector
+
+| Variable     | Required | Type    | Description
+| ------------ | -------- | ------- | -----------
+| device_id    | Yes      | String  | The device hex code of SR-IoV device. Allowed value "0d58", "1572", "158b", "1013", "1015", "1017", "101b".
+| net_filter   | No       | String  | Infrastructure Networking selection filter. Allowed value "openstack/NekID:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
```
[the remark this answers is not in the document]
Comment (tonyskapunk) [reply roles/sriov_config/README.md:56]: Ah! good catch, thanks!

File: roles/sriov_config/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tonyskapunk) [reply roles/sriov_config/README.md]: Oh, I did, in the template, the doc was wrong, that has been fixed, thanks to you! :+1:

**Comment by tonyskapunk** [comment]
Test with Mellanox: https://www.distributed-ci.io/jobs/ff235c8a-b950-4032-a099-7097eaa36750
Test with Intel: https://www.distributed-ci.io/jobs/03820227-4308-4e49-bc40-203e77811d76

**Review by ramperher — APPROVED** [review]
LGTM

**Review by manurodriguez — APPROVED** [review]
Thanks

**Pull request merged**